File: app/src/main/java/com/cv/anandmjoseph/anandcv/adapter/ExperienceAdapter.java
package com.cv.anandmjoseph.anandcv.adapter;

import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.cv.anandmjoseph.anandcv.R;
import com.cv.anandmjoseph.anandcv.customwidget.RobotoBoldTextView;
import com.cv.anandmjoseph.anandcv.customwidget.RobotoLightTextView;
import com.cv.anandmjoseph.anandcv.customwidget.RobotoMediumEditText;
import com.cv.anandmjoseph.anandcv.customwidget.RobotoMediumTextView;
import com.cv.anandmjoseph.anandcv.data.ExperienceData;

import java.util.List;

/**
 * Created by Anand M Joseph on 1/6/2018.
 */

public class ExperienceAdapter extends RecyclerView.Adapter<ExperienceAdapter.ExperienceViewHolder>{

    private List<ExperienceData> mExperienceDatalist;

    public static class ExperienceViewHolder extends RecyclerView.ViewHolder {
        CardView cv;
        TextView company_name;
        TextView company_period,company_desiganation;
        TextView company_role;
        ImageView personPhoto;

        ExperienceViewHolder(View itemView) {
            super(itemView);
           cv=(CardView)itemView.findViewById(R.id.exp_cv);
            personPhoto=(ImageView)itemView.findViewById(R.id.thumbnail);
            company_name=(RobotoBoldTextView)itemView.findViewById(R.id.company_name);
            company_period=(RobotoMediumTextView)itemView.findViewById(R.id.company_period);
            company_desiganation=(RobotoMediumTextView)itemView.findViewById(R.id.company_desiganation);
            company_role=(RobotoLightTextView)itemView.findViewById(R.id.company_role);
        }
    }

    public ExperienceAdapter(List<ExperienceData> data){
        this.mExperienceDatalist = data;
    }

    @Override
    public int getItemCount() {
        return mExperienceDatalist.size();
    }

    @Override
    public ExperienceViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.content_experince_iteam, viewGroup, false);
        ExperienceViewHolder expvh = new ExperienceViewHolder(v);
        return expvh;
    }

    @Override
    public void onBindViewHolder(ExperienceViewHolder holder, int position) {
        ExperienceData mExperienceData=mExperienceDatalist.get(position);
       holder.company_name.setText(mExperienceData.getCompany_name());
       holder.company_period.setText(mExperienceData.getWorking_period());
       holder.company_role.setText(mExperienceData.getCompany_desiganation());
       holder.company_desiganation.setText(mExperienceData.getRole_detail());
       holder.personPhoto.setImageResource(mExperienceData.getCompany_photoId());
    }

}

File: app/src/main/java/com/cv/anandmjoseph/anandcv/adapter/ExperienceAdapter.kt
package com.cv.anandmjoseph.anandcv.adapter

import android.support.v7.widget.CardView
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.cv.anandmjoseph.anandcv.R
import com.cv.anandmjoseph.anandcv.adapter.ExperienceAdapter.ExperienceViewHolder
import com.cv.anandmjoseph.anandcv.customwidget.RobotoBoldTextView
import com.cv.anandmjoseph.anandcv.customwidget.RobotoLightTextView
import com.cv.anandmjoseph.anandcv.customwidget.RobotoMediumEditText
import com.cv.anandmjoseph.anandcv.customwidget.RobotoMediumTextView
import com.cv.anandmjoseph.anandcv.data.ExperienceData

/**
 * Created by Anand M Joseph on 1/6/2018.
 */

class ExperienceAdapter(private val mExperienceDatalist: List<ExperienceData>) : RecyclerView.Adapter<ExperienceAdapter.ExperienceViewHolder>() {

    class ExperienceViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var cv: CardView
        internal var company_name: TextView
        internal var company_period: TextView
        internal var company_desiganation: TextView
        internal var company_role: TextView
        internal var personPhoto: ImageView

        init {
            cv = itemView.findViewById<View>(R.id.exp_cv) as CardView
            personPhoto = itemView.findViewById<View>(R.id.thumbnail) as ImageView
            company_name = itemView.findViewById<View>(R.id.company_name) as RobotoBoldTextView
            company_period = itemView.findViewById<View>(R.id.company_period) as RobotoMediumTextView
            company_desiganation = itemView.findViewById<View>(R.id.company_desiganation) as RobotoMediumTextView
            company_role = itemView.findViewById<View>(R.id.company_role) as RobotoLightTextView
        }
    }

    override fun getItemCount(): Int {
        return mExperienceDatalist.size
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ExperienceViewHolder {
        val v = LayoutInflater.from(viewGroup.context).inflate(R.layout.content_experince_iteam, viewGroup, false)
        return ExperienceViewHolder(v)
    }

    override fun onBindViewHolder(holder: ExperienceViewHolder, position: Int) {
        val mExperienceData = mExperienceDatalist[position]
        holder.company_name.setText(mExperienceData.company_name)
        holder.company_period.setText(mExperienceData.working_period)
        holder.company_role.setText(mExperienceData.company_desiganation)
        holder.company_desiganation.setText(mExperienceData.role_detail)
        holder.personPhoto.setImageResource(mExperienceData.company_photoId)
    }

}